### ISSUE TITLE
Error when dynamically attaching specialagent jar- Java 8

### ISSUE THREAD

**Issue by khiradeshubham** [opentracing-contrib/java-specialagent#99]
Hello, I am getting the following error when I'm dynamically Attaching _specialagent_ to my application

here is what I got in application logs

```
java.lang.reflect.InvocationTargetException failed to start!
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
        at sun.instrument.InstrumentationImpl.loadClassAndCallAgentmain(InstrumentationImpl.java:411)
Caused by: java.lang.IllegalStateException: java.util.zip.ZipException: error in opening zip file
        at io.opentracing.contrib.specialagent.SpecialAgentUtil.findJarResources(SpecialAgentUtil.java:337)
        at io.opentracing.contrib.specialagent.SpecialAgent.initialize(SpecialAgent.java:213)
        at io.opentracing.contrib.specialagent.ByteBuddyManager.premain(ByteBuddyManager.java:102)
        at io.opentracing.contrib.specialagent.SpecialAgent.premain(SpecialAgent.java:165)
        at io.opentracing.contrib.specialagent.SpecialAgent.agentmain(SpecialAgent.java:176)
        ... 6 more
Caused by: java.util.zip.ZipException: error in opening zip file
        at java.util.zip.ZipFile.open(Native Method)
        at java.util.zip.ZipFile.<init>(ZipFile.java:225)
        at java.util.zip.ZipFile.<init>(ZipFile.java:155)
        at java.util.jar.JarFile.<init>(JarFile.java:166)
        at java.util.jar.JarFile.<init>(JarFile.java:130)
        at io.opentracing.contrib.specialagent.SpecialAgentUtil$Plugin.getPlugin(SpecialAgentUtil.java:348)
        at io.opentracing.contrib.specialagent.SpecialAgentUtil$Plugin.access$000(SpecialAgentUtil.java:341)
        at io.opentracing.contrib.specialagent.SpecialAgentUtil.findJarResources(SpecialAgentUtil.java:292)
        ... 10 more

```
and this with dynamic attachment,  

```
Exception in thread "main" com.sun.tools.attach.AgentInitializationException: Agent JAR loaded but agent failed to initialize
	at sun.tools.attach.HotSpotVirtualMachine.loadAgent(HotSpotVirtualMachine.java:121)
	at io.opentracing.contrib.specialagent.SpecialAgent.main(SpecialAgent.java:148)
```

Java 
`java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)`

My application is a Guice+undertow

**Comment by ledor473** [comment]
@khiradeshubham can you provide the full command you used to add the agent? Ideally, include the full name of the `.jar` you've used.

**Comment by khiradeshubham** [comment]
Hello @ledor473 , 
I tried 
`java -Xbootclasspath/a:/Library/Java/JavaVirtualMachines/jdk1.8.0_181.jdk/Contents/Home/lib/tools.jar -jar opentracing-specialagent-1.2.0.jar <pid>`
and renamed version as well
`java -Xbootclasspath/a:/Library/Java/JavaVirtualMachines/jdk1.8.0_181.jdk/Contents/Home/lib/tools.jar -jar opentracing-specialagent-1.2.0-SNAPSHOT.jar <pid>`
but same result.
Thanks

**Comment by SevaSafris** [comment]
Hi @khiradeshubham, I created a test app running OkHttp in a loop. I was able to attach to to the test app's PID with SpecialAgent via dynamic attach. If you could provide more information regarding your use-case, it would be very helpful to determine what's happening.

**Comment by khiradeshubham** [comment]
Hello @SevaSafris, I'm using Gradle wrapper to start my application `gradlew run`, did not add anything specific to open tracing in source code and just attaching the SpecialAgent to that application process. Can you share your test application for reference?

**Comment by SevaSafris** [comment]
@khiradeshubham, here is the code I used for my test:

```java
package io.opentracing.contrib.specialagent.okhttp;

import static org.junit.Assert.*;

import java.io.IOException;
import java.lang.management.ManagementFactory;

import io.opentracing.Tracer;
import io.opentracing.util.GlobalTracer;
import okhttp3.HttpUrl;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.mockwebserver.MockResponse;
import okhttp3.mockwebserver.MockWebServer;

public class Main {
  public static int getPID() {
    final String pidAtHost = ManagementFactory.getRuntimeMXBean().getName();
    if (pidAtHost == null)
      return -1;

    try {
      return Integer.parseInt(pidAtHost.substring(0, pidAtHost.indexOf('@')));
    }
    catch (final NumberFormatException e) {
      return -1;
    }
  }

  public static void main(final String[] args) throws InterruptedException, IOException {
    System.out.println(System.getProperty("java.class.path"));
    System.out.println(getPID());
    System.out.println("You have 5 seconds to attach...");
    Thread.sleep(5000);
    test();
  }

  private static void test() throws InterruptedException, IOException {
    try (final MockWebServer server = new MockWebServer()) {
      while (true) {
        try (final Tracer tracer = GlobalTracer.get()) {
          final OkHttpClient client = new OkHttpClient();
          try {
            server.enqueue(new MockResponse().setBody("hello, world!").setResponseCode(200));

            final HttpUrl httpUrl = server.url("/hello");

            final Request request = new Request.Builder().url(httpUrl).build();
            final Response response = client.newCall(request).execute();

            assertTrue(200 == response.code());
            assertTrue(1 == client.interceptors().size());
            assertTrue(1 == client.networkInterceptors().size());
            System.out.println("Ok");
          }
          catch (final Throwable e) {
            System.err.println(e.getClass().getSimpleName() + ": " + e.getMessage());
          }

          Thread.sleep(1000);
        }
      }
    }
  }
}
```

The required dependencies for this test come from `rules/specialagent-okhttp/pom.xml`:

```xml
  <dependencies>
    <dependency>
      <groupId>com.squareup.okhttp3</groupId>
      <artifactId>okhttp</artifactId>
      <version>3.6.0</version>
    </dependency>
    <dependency>
      <groupId>com.squareup.okhttp3</groupId>
      <artifactId>mockwebserver</artifactId>
      <version>3.6.0</version>
    </dependency>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>4.12</version>
    </dependency>
  </dependencies>
```

**Comment by SevaSafris** [comment]
Hi @khiradeshubham, were you able to resolve the problems you were encountering? Though not directly related, I have resolved an issue with propagation of system arguments for invocation via dynamic attach. This has been pushed and released to the current development version: 1.3.2-SNAPSHOT.

**Comment by SevaSafris** [comment]
Hi @khiradeshubham, please advise whether you are still experiencing issues with dynamic attach. If not, I will close this issue.
